interlok-core/src/main/java/com/adaptris/core/management/config/SharedConnectionConfigurationChecker.java

```diff
@@ -0,0 +1,125 @@
+      
+      NodeList referencedProduceConnections = scanForConnections(xmlDocument, XPATH_REFERENCED_PRODUCE_CONNECTIONS);
+      System.err.println("Found " + referencedProduceConnections.getLength() + " referenced produce connection(s).");
+      
+      boolean warningsFound = false;
+      StringBuilder warningText = new StringBuilder();
+      
+      // **********************************
+      // Check all shared connections are used.
+      for(int counter = 0; counter < availableConnections.getLength(); counter ++) {
+        if(!this.testReferenceExistsInNodeSets(availableConnections.item(counter).getTextContent(), referencedConsumeConnections)) {
+          warningsFound = true;
+          warningText.append("\nShared connection unused: " + availableConnections.item(counter).getTextContent());
+        }
+      }
+      
+      // **********************************
+      // Check all consume shared connections exist.
+      for(int counter = 0; counter < referencedConsumeConnections.getLength(); counter ++) {
+        if(!this.testReferenceExistsInNodeSets(referencedConsumeConnections.item(counter).getTextContent(), availableConnections)) {
+          warningsFound = true;
+          warningText.append("\nConsume shared connection does not exist in shared connections: " + referencedConsumeConnections.item(counter).getTextContent());
+        }
+      }
+      
+      // **********************************
+      // Check all produce shared connections exist.
+      for(int counter = 0; counter < referencedProduceConnections.getLength(); counter ++) {
+        if(!this.testReferenceExistsInNodeSets(referencedProduceConnections.item(counter).getTextContent(), availableConnections)) {
+          warningsFound = true;
+          warningText.append("\nProduce shared connection does not exist in shared connections: " + referencedProduceConnections.item(counter).getTextContent());
+        }
+      }
+      
+      report.setCheckPassed(!warningsFound);
+      if(warningsFound)
+        throw new ConfigurationException(warningText.toString());
+      
+    } catch (Exception ex) {
+      report.setFailureException(ex);
+    }
+    
+    return report;
+  }
+  
+  private boolean testReferenceExistsInNodeSets(String searchValue, NodeList listOfNodes) {
+    boolean result = false;
+    for(int counter = 0; counter < listOfNodes.getLength(); counter ++) {
+      if(searchValue.equals(listOfNodes.item(counter).getTextContent())) {
+        result = true;
+        break;
+      }
+    }
+    
+    return result;
+  }
+
+  private Document buildXmlDocument(BootstrapProperties bootProperties) throws Exception {
+    InputStream configuration = bootProperties.getConfigurationStream();
+    
+    DocumentBuilderFactory builderFactory = DocumentBuilderFactory.newInstance();
+    builderFactory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+    DocumentBuilder builder = builderFactory.newDocumentBuilder();
```

Comment:
DocumentBuilderFactoryBuilder.newRestrictedInstance().build() -> does it all for you, since you have forgotten the ExpandEntityReferences=false so you are now possibly vulnerable to the

<!ENTITY % StandardInfo SYSTEM "/etc/passwd">